game/scripts/labels/quiz_session.rpy

```diff
@@ -26,7 +27,11 @@ label study_session:
             $ quiz_question = renpy.random.choice(easter_egg_quiz_questions)
             $ renpy.notify("A wild Easter Egg question has appeared!")
         else: # regular study session questions
-            $ quiz_question = renpy.random.choice(study_session_questions)
+            $ quiz_question = None
+            # only show a question not seen in the same session
+            while not quiz_question or quiz_question.question in questions_asked:
+                $ quiz_question = renpy.random.choice(study_session_questions)
+            $ questions_asked.add(quiz_question.question)
```

Comment:
Hi @neil-sriv if I'm reading this right, this will loop forever once a user has seen all the questions.  An alternative approach could be to shuffle the question order, loop over them all and then, once you're out of questions, shuffle again.

Reply:
Hey @ojeytonwilliams, thanks for taking the time to review my PR. I did test the code and never got into a infinite loop. Because the question limit is hardcoded at 4, it would be impossible for the `questions_asked` set to contain every question in `study_session_questions`. The `questions_asked` set is created on every quiz session and does not contain the questions from the previous session.

Reply:
Ah, thanks for explaining - that makes perfect sense.

I'll leave this for Lynn to merge, but this LGTM.